library/cwakuv2.nim

```diff
@@ -0,0 +1,160 @@
+
+import
+  std/[sequtils,times,strformat,json],
+  os
+import
+  chronicles,
+  chronos
+import
+  ../../waku/v2/waku_core/message/codec,
+  ../../waku/v2/waku_core/message/message,
+  ../../waku/v2/waku_core/topics/pubsub_topic,
+  ../../apps/wakunode2/wakunode2,
+  events/[json_error_event,json_message_event,json_signal_event]
+
+################################################################################
+### Not exported components
+
+type
+  EventCallback = proc(signal: cstring) {.cdecl, gcsafe, raises: [Defect].}
+
+var eventCallback:EventCallback = nil
+
+proc callbackHandler(pubsubTopic: string, data: seq[byte]): Future[void] {.gcsafe, raises: [Defect].} =
+  if eventCallback != nil:
+    let msg = WakuMessage.decode(data)
+    var event: JsonSignal
+    if msg.isOk():
+      event = JsonMessageEvent.new(pubsubTopic, msg.value)
+    else:
+      let errorMsg = string("Error decoding message.") & $msg.error
+      event = JsonErrorEvent.new(errorMsg)
+
+    try:
+      eventCallback(cstring($event))
+    except Exception:
+      error "Exception when calling 'eventCallBack': " &
+            getCurrentExceptionMsg()
+  else:
+    error "eventCallback is nil"
+
+  var retFut = newFuture[void]()
+  retFut.complete()
+  return retFut
+
+proc okResp(message: string): string =
+  $(%* { "result": message })
+
+proc errResp(message: string): string =
+  # Convers an error message into an error-JsonResponse
+  # {
+  #   error: string;
+  # }
+  return $(%* { "error": message })
+
+################################################################################
+### Exported components
+
+proc show_version(): cstring {.dynlib, exportc.} =
+  return wakuNode2VersionString
```

Comment:
Nice!

Reply:
I think this can be `waku_version()`. `waku_node` already provides `git_version`, so I don't think this needs to be exported elsewhere. If we need to construct something specific, that should be done on the Nim API in `waku_node.nim`.